Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -138,19 +138,31 @@ void WarpXOpenPMDPlot::GetFileName(std::string& filename)
 }
 
 
-void WarpXOpenPMDPlot::SetStep(int ts)
+bool WarpXOpenPMDPlot::SetStep(int ts)
 {
   if (ts < 0)
-    return;
-
-  m_CurrentStep =  ts;
-
-  Init(openPMD::AccessType::CREATE);
+    return false;
+
+  if (m_CurrentStep >= ts) {
+     m_OK = false;
+  } else {
+    m_OK = true;
+    m_CurrentStep =  ts;
+    Init(openPMD::AccessType::CREATE);
+  }
+  return  m_OK;
 }
 
 void
 WarpXOpenPMDPlot::Init(openPMD::AccessType accessType)
 {
+  // closing the previous m_Series is needed.
+  // adios1 will crash for example.
```

Comment:
Hm, this is confusing to me.
Overwriting a unique ptr will free the old value it holds. this will then trigger the `Series` destructor which should then `flush()`, too.

Reply:
Okay, after further tests, it is now clear that this is only an ADIOS1 upstream bug:
https://github.com/openPMD/openPMD-api/pull/685
https://github.com/openPMD/openPMD-api/pull/686

nevertheless, I think we can simplify the code herein to:
```c++
// close a previously open series before creating a new one
if (m_Series != nullptr)
    m_Series = nullptr;
```